server/api/core/core.js

```diff
@@ -394,6 +394,9 @@ export default {
       limit: 1,
       fields: {
         _id: 1
+      },
+      orderby: {
+        merchantShops: -1 // give preference to PrimaryShop if multiple results
```

Comment:
if this were SQL, I would have done something like:
``` SQL
SELECT _id
FROM shops
WHERE domain = 'domain'
ORDER BY CASE shopType WHEN 'primary' THEN 0 ELSE 1 END # <--
LIMIT 1
```

but I couldn't find an equivalent construct in Mongo.
Instead, I sort on whether a Shop has Merchant Shops, assuming only the Primary Shop does.

Reply:
@pmn4 Since `merchantShops` is an array, I wonder if this is going to always give you the results you want? Seems like we would want to sort on `merchantShop.lengrth` somehow?

Reply:
Wow, we call this function > 900 times on startup.

Reply:
we sure do! and about 15/request (I have [a PR](https://github.com/reactioncommerce/reaction/pull/3510) for that too!)

Reply:
as for the sort, it is definitely not ideal. It is basically a null/not-null sort.
seems like sorting on length would be an expensive query, but since you are not likely to have too many shops, I guess it would be ok.
I am definitely out of my depth when it comes to Mongo, so open to any suggestions

Reply:
Since `orderBy` is basically happening **after** the find, I wonder how much slower sorting it in memory would be? Although I wonder why this is doing a `find` and then a `fetch[0]`. Seems like we could be doing a `findOne`? If there's 1000 shops in a marketplace, that's a lot of data to pull across the network and then immediately throw away?

Reply:
there is a `limit: 1` in the `find`, so it shouldn't be pulling much/any more data across the wire than `findOne` would. Doesn't seem like getting a cursor here helps at all (since we immediately get the first record), so happy to change it to `findOne`